pkg/api/server/v1alpha2/result/result.go

```
@@ -31,28 +33,37 @@ func ParseName(raw string) (parent, name string, err error) {
 
 // ToStorage converts an API Result into its corresponding database storage
 // equivalent.
-// parent,name should be the name parts (e.g. not containing "/results/").
 func ToStorage(r *pb.Result) (*db.Result, error) {
 	parent, name, err := ParseName(r.GetName())
 	if err != nil {
 		return nil, err
 	}
+	ann, err := json.Marshal(r.GetAnnotations())
+	if err != nil {
+		return nil, err
+	}
 	result := &db.Result{
 		Parent:      parent,
 		ID:          r.GetId(),
 		Name:        name,
 		UpdatedTime: r.UpdatedTime.AsTime(),
 		CreatedTime: r.CreatedTime.AsTime(),
+		Annotations: ann,
 	}
 	return result, nil
 }
 
 // ToAPI converts a database storage Result into its corresponding API
 // equivalent.
 func ToAPI(r *db.Result) *pb.Result {
+	ann := map[string]string{}
+	if err := json.Unmarshal(r.Annotations, &ann); err != nil {
+		ann = nil
```

We should return the error here.

---

Done